src/dialects/postgres/index.js

```diff
@@ -199,15 +199,15 @@ assign(Client_PG.prototype, {
   _stream(connection, obj, stream, options) {
     const PGQueryStream = process.browser ? undefined : require('pg-query-stream');
     const sql = obj.sql;
+
     return new Promise(function(resolver, rejecter) {
       const queryStream = connection.query(new PGQueryStream(sql, obj.bindings, options));
-      queryStream.on('error', function(error) { stream.emit('error', error); });
-      // 'error' is not propagated by .pipe, but it breaks the pipe
-      stream.on('error', function(error) {
-        // Ensure the queryStream is closed so the connection can be released.
-        queryStream.close();
```

Comment:
Why was any of this changed? This query still needs to be closed on `error` or it will continue to execute on the server. See https://github.com/tgriesser/knex/pull/1935

Reply:
Good catch! @fcmatteo Any idea why queryStream.close() was dropped?

Reply:
Ouch. Can we get a test for this?..

Reply:
Test case for original fix would have catch this that is the reason I'm so insistent that all the fixes has test included. So when this is fixed again, a test would be great to have.

Reply:
@kibertoad you were faster 👍

Reply:
There are tests in the original PR, actually, I wonder if this particular aspect can be properly tested.

Reply:
Maybe by mocking queryStream.close method or by checking that stream.on('error'... event handler is set.

Reply:
Or with integration test which queries for open connections from database.

Reply:
I don't remember. Following the thread of the PR, there were two failing tests in PostgreSQL that apparently were fixed by that.
But testing on local with the previous chunk of code seems to work too.

Reply:
Ok, better to just put it back then and add new test

Reply:
Followed up with https://github.com/tgriesser/knex/pull/2773